components/director/internal/open_resource_discovery/validation_test.go

```diff
@@ -3304,6 +3581,31 @@ func TestDocuments_ValidateVendor(t *testing.T) {
 				doc := fixORDDocument()
 				doc.Vendors[0].Labels = json.RawMessage(invalidLabelsWhenKeyIsWrong)
 
+				return []*open_resource_discovery.Document{doc}
+			},
+		},
+		{
+			Name: "Invalid JSON object `Partners` field for Vendor",
+			DocumentProvider: func() []*open_resource_discovery.Document {
+				doc := fixORDDocument()
+				doc.Vendors[0].Partners = json.RawMessage(`[]`)
+
+				return []*open_resource_discovery.Document{doc}
+			},
+		}, {
+			Name: "`Partners` values are not array for Vendor",
+			DocumentProvider: func() []*open_resource_discovery.Document {
+				doc := fixORDDocument()
+				doc.Vendors[0].Labels = json.RawMessage(invalidPartnersWhenValueIsNotArray)
+
+				return []*open_resource_discovery.Document{doc}
+			},
+		}, {
+			Name: "`Partners` values are not array of strings for Vendor",
+			DocumentProvider: func() []*open_resource_discovery.Document {
+				doc := fixORDDocument()
+				doc.Vendors[0].Labels = json.RawMessage(invalidPartnersWhenValuesAreNotArrayOfStrings)
```

Comment:
Two things here:

1. `doc.Vendors[0].Partners`
2. the `invalidPartnersWhenValuesAreNotArrayOfStrings` JSON is not correct for the use-case of that test. The json should be `["<validPartner>", 31231]` i.e a validPartner and a number